PocketKit/Sources/PocketKit/Main/RegularMainCoordinator.swift

```diff
@@ -148,14 +148,16 @@ class RegularMainCoordinator: NSObject {
         }.store(in: &subscriptions)
 
         // HOME
-        model.home.$selectedReadableViewModel.receive(on: DispatchQueue.main).sink { [weak self] readable in
-            if readable != nil {
-                self?.model.home.selectedSlateDetailViewModel?.selectedReadableViewModel = nil
-                self?.model.myList.savedItemsList.selectedItem = nil
-                self?.model.myList.archivedItemsList.selectedItem = nil
+        model.home.$selectedReadableType.receive(on: DispatchQueue.main).sink { [weak self] readableType in
+            switch readableType {
+            case .recommendation(let viewModel):
+                self?.show(viewModel)
+            case .savedItem(let viewModel):
+                self?.show(viewModel)
+            case .none:
+                return
```

Comment:
do we also want to do`readerSubscriptions = []` here like we do in `CompactHomeCoordinator`?

Reply:
good catch!